### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by tertsdiepraam**

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
This is the same CI as uutils-term-grid. If this works I want to release this crate.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        # display coverage files
+        grcov . --output-type files --ignore build.rs --ignore "vendor/*" --ignore "/*" --ignore "[a-zA-Z]:/*" --excl-br-line "^\s*((debug_)?assert(_eq|_ne)?!|#\[derive\()" | sort --unique
+        # generate coverage report
+        grcov . --output-type lcov --output-path "${COVERAGE_REPORT_FILE}" --branch --ignore build.rs --ignore "vendor/*" --ignore "/*" --ignore "[a-zA-Z]:/*" --excl-br-line "^\s*((debug_)?assert(_eq|_ne)?!|#\[derive\()"
+        echo "name=report::${COVERAGE_REPORT_FILE}" >> $GITHUB_OUTPUT
+    - name: Upload coverage results (to Codecov.io)
+      uses: codecov/codecov-action@v4
+      # if: steps.vars.outputs.HAS_CODECOV_TOKEN
+      with:
+        # token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (cakebaker) [inline .github/workflows/ci.yml]: `codevov` requires a token
```suggestion
        token: ${{ secrets.CODECOV_TOKEN }}
```

Comment (tertsdiepraam) [reply .github/workflows/ci.yml]: Ah so I need to enable the `if` line above that as well?

Comment (cakebaker) [reply .github/workflows/ci.yml]: I don't think it is necessary because the token should always be there as it has to be added to Github's settings.

**Comment by tertsdiepraam** [comment]
@cakebaker ah, so now we need the codecov token. How do we do that?

**Comment by cakebaker** [comment]
@tertsdiepraam we have to get it from codecov.io and then add it as a secret on github, see https://github.com/codecov/codecov-action/?tab=readme-ov-file#usage

**Comment by tertsdiepraam** [comment]
Alright, I think I fixed it :)

**Review by cakebaker — APPROVED** [review]

**Pull request merged**